x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/utils/indices.ts

```diff
@@ -36,6 +36,16 @@ export function isConnectorIndex(
   );
 }
 
+export function isConnectorCrawlerIndex(
+  index: ElasticsearchIndexWithIngestion | undefined
+): index is ConnectorIndex {
```

Comment:
I've updated it to return `CrawlerIndex`

Reply:
Yes, because now crawler isn't guaranteed to have a connector property! But any crawler that goes through here is.

What we really need is a CrawlerConnector type that always has a defined connector property.